library/dnet.c

```diff
@@ -783,6 +785,25 @@ int dnet_server_send_write(struct dnet_server_send_ctl *send,
 			(unsigned long long)re->iterated_keys, (unsigned long long)re->total_keys,
 			atomic_read(&send->bytes_pending), send->bytes_pending_max);
 
+	// check that source and destination backends are not the same
```

Comment:
Shouldn't it be on the client side?
Do not even try to start iterator if its source/destination backends are the same?